app/services/UnloadingPermissionPDFService.scala

```diff
@@ -28,33 +30,38 @@ import uk.gov.hmrc.http.HeaderCarrier
 import javax.inject.Inject
 import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
-import models.PdfDocument
+import scala.xml.NodeSeq
 
 class UnloadingPermissionPDFService @Inject()(messageRetrievalService: MessageRetrievalService, manageDocumentsConnector: ManageDocumentsConnector)
     extends Results {
 
   val explicitHeaders = Set(CONTENT_LENGTH, CONTENT_TYPE, CONTENT_DISPOSITION)
 
+  private def generatePDF(messageBody: NodeSeq)(implicit hc: HeaderCarrier, ec: ExecutionContext) =
+    manageDocumentsConnector.getUnloadingPermissionPdf(messageBody).map {
+      result =>
+        result.status match {
+          case OK =>
+            Right(
+              PdfDocument(
+                result.bodyAsSource,
+                result.header(CONTENT_LENGTH).map(_.toLong),
+                result.header(CONTENT_TYPE),
+                result.header(CONTENT_DISPOSITION)
+              )
+            )
+          case otherErrorCode =>
+            Left(OtherError(otherErrorCode, result.body))
+        }
+    }
+
   def getPDF(arrival: Arrival)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[WSError, PdfDocument]] =
     messageRetrievalService.getUnloadingPermission(arrival) match {
-      case Some(unloadingPermission) =>
-        manageDocumentsConnector.getUnloadingPermissionPdf(unloadingPermission).map {
-          result =>
-            result.status match {
-              case OK =>
-                Right(
-                  PdfDocument(
-                    result.bodyAsSource,
-                    result.header(CONTENT_LENGTH).map(_.toLong),
-                    result.header(CONTENT_TYPE),
-                    result.header(CONTENT_DISPOSITION)
-                  )
-                )
-              case otherErrorCode =>
-                Left(OtherError(otherErrorCode, result.body))
-            }
-        }
-      case None => Future.successful(Left(NotFoundError))
+      case Some(unloadingPermission) => generatePDF(unloadingPermission.message)
+      case None                      => Future.successful(Left(NotFoundError))
     }
 
+  def getPDF(messages: List[MovementMessage])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[WSError, PdfDocument]] =
+    generatePDF(messages.minBy(_.messageCorrelationId).message)
```

Comment:
I think we would usually get the `.maxBy(_.messageCorrelationId)` i.e. the most recent message?

Reply:
I took this from https://github.com/hmrc/transit-movements-trader-at-destination/blob/main/app/services/ArrivalMessageSummaryService.scala#L102, which suggests that we get the first message rather than the last. Is it possible that this in itself is a bug? I'd imagine it probably is because everything else is max rather than min.

Might be good for me to introduce another test in here to back up what the right behaviour is.

Reply:
Resolving as this seems to be deliberate https://jira.tools.tax.service.gov.uk/browse/CTDA-1144